server/routes/notification.js

```diff
@@ -0,0 +1,244 @@
+const authMiddleware = require("../middlewares/authMiddleware");
+const ListProductModel = require("../models/listProductModel");
+const NotificationModel = require("../models/notificationModel");
+const { sequelize } = require("../models/productModel");
+const ProductModel = require("../models/productModel");
+const UserListModel = require("../models/userListModel");
+const { PRICE, ALL_TYPES_OBJECT } = require("../utils/enums");
+const router = require("express").Router();
+
+//Make the given notification id read
+router.put("/read/:id", authMiddleware, async (req, res) => {
+  try {
+    const { id: notificationId } = req.params;
+    const { id: userId } = req.user;
+
+    const notification = await NotificationModel.findOne({
+      where: {
+        id: notificationId,
+        userId: userId,
+      },
+    });
+
+    if (!notification)
+      return res.status(400).send({
+        error: {
+          msg: "Notification not found.",
+        },
+      });
+
+    notification.isRead = true;
+
+    const updatedNotification = await notification.save();
+
+    res.send(updatedNotification);
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({
+      error: {
+        msg: "Server Error",
+        data: err,
+      },
+    });
+  }
+});
+
+//Paginated querying of notifications of a user
+//Accepts "page" as a query. By default will be a 1.
+//Accepts "order" as a query. By default is "DESC" to show latest notifs first
+//Accepts "type" as a query. By default is an array of all types to show all types of notifications
+//Accepts "isRead" as a query. By default is false to show only unread notifs
+//Accepts "maxNotifications" as a query. By default is 10
+router.get("/get-notifications", authMiddleware, async (req, res) => {
+  const { id: userId } = req.user;
+  let { order, page, maxNotifications, type, isRead } = req.query;
+
+  const typesArray = Object.values(ALL_TYPES_OBJECT);
+
+  //#region Query Validation
+
+  if (!type) {
+    type = typesArray;
+  } else {
+    type = type.toLowerCase();
+    if (!ALL_TYPES_OBJECT[type.toUpperCase()]) {
+      return res
+        .status(400)
+        .send({ error: { msg: "Invalid type query", errorCode: 400 } });
+    }
+  }
+
+  if (!page) {
+    page = 1;
+  } else {
+    if (isNaN(parseInt(page))) {
+      return res
+        .status(400)
+        .send({ error: { msg: "Invalid page query", errorCode: 400 } });
+    }
+  }
+
+  if (!maxNotifications) {
```

Comment:
Maybe this is something to talk with @sina-jamshidi , but I think it would make more sense to move the whole query validation/transformation logic to a separate function/middleware.

Reply:
Yeah if you can modularize it a bit more even by moving them inside a function and returning an errorCode/message that would be good